dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/Matrix.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.blas;
+
+/**
+ * Matrix interface.
+ */
+public interface Matrix {
+
+  /**
+   * Returns the number of rows.
+   */
+  int getRows();
+
+  /**
+   * Returns the number of columns.
+   */
+  int getColumns();
+
+  /**
+   * Returns a element specified by the given index (linear indexing).
+   */
+  float get(final int i);
+
+  /**
+   * Returns elements specified by the linear indices.
+   */
+  Matrix get(final int[] indices);
+
+  /**
+   * Returns a element specified by the row and column indices.
+   */
+  float get(final int rowIndex, final int columnIndex);
+
+  /**
+   * Sets a matrix element (linear indexing).
+   */
+  Matrix put(final int i, final float v);
```

Comment:
The variable name `i` is inconsistent with `put(rowIndex, columnIndex, v)`. We should use `index` or `linearIndex`.